src/controllers/app_controller.go

```diff
@@ -0,0 +1,113 @@
+package controllers
+
+import (
+	"github.com/gin-gonic/gin"
+	"golang-bootcamp-2020/services"
+	_errors "golang-bootcamp-2020/utils/error"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type AppController interface {
+	FetchData(c *gin.Context)
+	GetHealth(c *gin.Context)
+	GetCharacterById(c *gin.Context)
+	GetCharacters(c *gin.Context)
+	GetCharacterIdByName(c *gin.Context)
+}
+
+type appController struct {
+	service services.Service
+}
+
+type idResponse struct {
+	Id string `json:"id"`
```

Comment:
By convention (most lints even show a warning) the `Id` fields should be renamed as `ID`